integration/deployment_test.go

```diff
@@ -0,0 +1,155 @@
+//go:build integration
+// +build integration
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/deployment"
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	namespacePrefix = "ecogoinfra-deployment"
+	containerImage  = "nginx:latest"
+	timeoutDuration = time.Duration(60) * time.Second
+)
+
+func TestDeploymentCreate(t *testing.T) {
+	t.Parallel()
+	client := clients.New("")
+
+	// Create a random namespace for the test
+	randomNamespace := generateRandomNamespace(namespacePrefix)
+
+	// Create the namespace in the cluster using the namespaces package
+	_, err := namespace.NewBuilder(client, randomNamespace).Create()
+	assert.Nil(t, err)
+
+	// Defer the deletion of the namespace
+	defer func() {
+		// Delete the namespace
+		err := namespace.NewBuilder(client, randomNamespace).Delete()
```

Comment:
Why do you use namespace.NewBuilder 1 more time? You can save previous object and re-use Delete function instead of init the same object.

Reply:
Good call, switched to a single builder style.